core/api/src/debug/add-payment-request-all-wallet-invoice.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * how to run:
+ *	. ./.envrc && yarn ts-node \
+ *		--files \
+ *			-r tsconfig-paths/register \
+ *			-r src/services/tracing.ts \
+ *		src/debug/add-payment-request-all-wallet-invoices.ts
+ */
+
+import { LndService } from "@/services/lnd"
+import { lndsConnect } from "@/services/lnd/auth"
+import { isUp } from "@/services/lnd/health"
+import { setupMongoConnection } from "@/services/mongodb"
+import {
+  SemanticAttributes,
+  addEventToCurrentSpan,
+  asyncRunInSpan,
+  wrapAsyncToRunInSpan,
+} from "@/services/tracing"
+import { WalletInvoice } from "@/services/mongoose/schema"
+
+const main = async () => {
+  return asyncRunInSpan(
+    "debug.addPaymentRequestAllWalletInvoices",
+    {
+      attributes: {
+        [SemanticAttributes.CODE_FUNCTION]: "addPaymentRequestAllWalletInvoices",
+        [SemanticAttributes.CODE_NAMESPACE]: "debug",
+      },
+    },
+    async () => {
+      const initialNumberOfLnInvoicesMissingPaymentRequest =
+        await WalletInvoice.countDocuments({
+          paymentRequest: { $exists: false },
+        })
+
+      console.log(
+        `Number of wallet invoices missing payment request: ${initialNumberOfLnInvoicesMissingPaymentRequest}`,
+      )
+
+      const lndService = LndService()
+      if (lndService instanceof Error) return lndService
+      const pubkeys = lndService.listActivePubkeys()
+      for (const pubkey of pubkeys) {
```

Comment:
note: we had recycle node in the past (following issue with our node), so there will paymentRequest missing

Reply:
Good to know